src/librados/IoCtxImpl.cc

```diff
@@ -1757,6 +1757,9 @@ int librados::IoCtxImpl::notify(const object_t& oid, bufferlist& bl,
   objecter->linger_cancel(linger_op);
 
   set_sync_op_version(objver);
+
+  delete notify_finish;
```

Comment:
i think either `notify_finish->complete(r);` on line 1757 or `Objecter::handle_watch_notify` will delete it.